### PR TITLE
Add markup_node support in wizard

### PR DESCRIPTION
**Pull request by artemipanchuk**

### CODE REVIEW

File: src/ui/units/wizard/containers/Wizard/SectionVisualization/PlaceholdersContainer/SegmentsPlaceholder/SegmentsPlaceholder.tsx [hunk 1]
```diff
@@ -54,7 +54,7 @@ class SegmentsPlaceholder extends React.Component<Props> {
     }
 
     handleCheckAllowed = (item: Field) => {
-        return isDimensionField(item);
+        return isDimensionField(item) && item.data_type !== DATASET_FIELD_TYPES.MARKUP;
```
Comment (kuzmadom) [inline src/ui/units/wizard/containers/Wizard/SectionVisualization/PlaceholdersContainer/SegmentsPlaceholder/SegmentsPlaceholder.tsx:57]: _isMarkupField_ ?

File: src/ui/constants/visualizations/metric.ts [hunk 2]
```diff
@@ -16,7 +17,14 @@ export const METRIC_VISUALIZATION: Shared['visualization'] = {
...
             allowedFinalTypes: ITEM_TYPES.MEASURES,
-            allowedDataTypes: PRIMITIVE_DATA_TYPES,
+            allowedDataTypes: PRIMITIVE_DATA_TYPES_AND_MARKUP,
+            checkAllowed: (item: Field) => {
+                const allowedDataTypes = Utils.isEnabledFeature(Feature.MarkupMetric)
+                    ? PRIMITIVE_DATA_TYPES_AND_MARKUP
+                    : PRIMITIVE_DATA_TYPES;
+
+                return allowedDataTypes.has(item.data_type);
+            },
```
Comment (kuzmadom) [inline src/ui/constants/visualizations/metric.ts:27]: It's a bit confusing - why are there the two fields with different types?  I think we should leave only one of the two

File: src/ui/units/wizard/utils/visualization.ts [hunk 3]
```diff
@@ -43,7 +44,11 @@ import {ITEM_TYPES} from '../constants';
 import {getCommonDataType} from './helpers';
 
 export const prepareFieldToMeasureTransformation = (item: Field): Field => {
-    if (ITEM_TYPES.MEASURES_AND_PSEUDO.has(item.type) || isParameter(item)) {
+    if (
+        ITEM_TYPES.MEASURES_AND_PSEUDO.has(item.type) ||
+        isParameter(item) ||
+        item.data_type === DATASET_FIELD_TYPES.MARKUP
```
Comment (kuzmadom) [inline src/ui/units/wizard/utils/visualization.ts:50]: Why is a markup added here?

File: src/ui/utils/visualizations/line.ts [hunk 4]
```diff
@@ -23,7 +24,11 @@ export function linearCheckColor({
     designItems,
     isMultipleColorsSupported,
 }: CommonCheckColorArgs) {
-    if (isMeasureField(item) || isMeasureValue(item)) {
+    if (
+        isMeasureField(item) ||
+        isMeasureValue(item) ||
+        item.data_type === DATASET_FIELD_TYPES.MARKUP
```
Comment (kuzmadom) [inline src/ui/utils/visualizations/line.ts:30]: Is this check really necessary here? Aren't the available field types checked elsewhere?

File: src/server/modes/charts/plugins/datalens/preparers/metric/index.ts [hunk 5]
```diff
@@ -1,92 +1,37 @@
...
+import {prepareBasicMetricVariant} from './variants/basic';
+import {prepareMarkupMetricVariant} from './variants/markup';
 
 function prepareMetric({placeholders, resultData, shared, idToTitle}: PrepareFunctionArgs) {
     const {data, order} = resultData;
 
     const measure = placeholders[0].items[0];
 
+    if (typeof measure === 'undefined') {
+        return {};
```
Comment (kuzmadom) [inline src/server/modes/charts/plugins/datalens/preparers/metric/index.ts:15]: Is this a valid return result? What will be displayed in this case?

File: src/server/modes/charts/plugins/datalens/preparers/metric/variants/markup.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+        return {
+            value: {
+                type: 'concat',
+                className: `markup-indicator_size_${size}`,
+                children: [
+                    {
+                        className: 'markup-indicator-title',
+                        type: 'text',
+                        content: measure.fakeTitle || measure.title,
```
Comment (kuzmadom) [inline src/server/modes/charts/plugins/datalens/preparers/metric/variants/markup.ts:28]: The chart title is a separate setting. I don't think it should be a part of the markup.

File: src/server/modes/charts/plugins/datalens/preparers/metric/index.ts (hunk 5 above)
Comment (artemipanchuk) [reply src/server/modes/charts/plugins/datalens/preparers/metric/index.ts:15]: Yes. Chartkit will display "No data"

File: src/ui/utils/visualizations/line.ts (hunk 4 above)
Comment (artemipanchuk) [reply src/ui/utils/visualizations/line.ts:30]: No, this exact function checks available fields for colors section

File: src/server/modes/charts/plugins/datalens/preparers/metric/variants/markup.ts (hunk 6 above)
Comment (artemipanchuk) [reply src/server/modes/charts/plugins/datalens/preparers/metric/variants/markup.ts:30]: This part code is intended to be a replacement for old metric. Old metric had title as a part of it. But i've added now correct setting pass, to support title from extraSettings.

**Review by kuzmadom — APPROVED** [review]

**Comment by artemipanchuk** [comment]
Related to https://github.com/datalens-tech/datalens/issues/92

**Pull request merged**